src/server/pkg/obj/microsoft_client.go

```diff
@@ -116,71 +125,80 @@ type microsoftWriter struct {
 	container  string
 	blob       string
 	blobClient storage.BlobStorageClient
+	buf        *bytes.Buffer
+	nBlocks    int
+	eg         errgroup.Group
+	err        error
```

Comment:
Why store the `err` on the struct here?

Reply:
It's used in `Write` to exit early if previous async calls to `writeBlock` have failed. Without it errors would still get returned in `Close` but you might waste of cycles / bandwidth pushing blocks that aren't going anywhere.

Reply:
I'll add a comment to this effect.